src/ray/raylet/agent_manager.h

```diff
@@ -32,96 +30,50 @@ typedef std::function<std::shared_ptr<boost::asio::deadline_timer>(std::function
                                                                    uint32_t delay_ms)>
     DelayExecutorFn;
 
-typedef std::function<std::shared_ptr<rpc::RuntimeEnvAgentClientInterface>(
-    const std::string &ip_address, int port)>
-    RuntimeEnvAgentClientFactoryFn;
-
-/// Callback that's called after runtime env is created.
-/// \param[in] successful Whether or not the creation was successful.
-/// \param[in] serialized_runtime_env_context Serialized context.
-/// \param[in] setup_error_message The error message if runtime env creation fails.
-/// It must be only set when successful == false.
-typedef std::function<void(bool successful,
-                           const std::string &serialized_runtime_env_context,
-                           const std::string &setup_error_message)>
-    GetOrCreateRuntimeEnvCallback;
-typedef std::function<void(bool successful)> DeleteRuntimeEnvIfPossibleCallback;
-
-class AgentManager : public rpc::AgentManagerServiceHandler {
+// Manages a separate "Agent" process. In constructor (or the `StartAgnet` method) it
+// starts a process with `agent_commands` plus some additional arguments.
+//
+// Raylet outlives agents. This means when Raylet exits, the agents must exit; on the
+// other hand, if the agent exits, the raylet may exit only if fate_shares = true. This is
+// implement by the dtor of AgentManager to kill the agent process.
+//
+// We typically start these agents:
+// - The DashboardAgent: `ray/dashboard/agent.py`
+// - The RuntimeEnvAgent: `ray/_private/runtime_env/agent/main.py`
+class AgentManager {
  public:
   struct Options {
     const NodeID node_id;
+    const std::string agent_name;
+    // Commands to start the agent. Note we append extra arguments:
```

Comment:
```suggestion
    // Commands to start the agent. Note we append extra arguments for dashboard agent:
```

Reply:
moved the comment to the options factory since that's where we added this argument.